src/components/Network/TopologyList.tsx

```diff
@@ -1,31 +1,167 @@
-import React, { useMemo } from 'react'
+import React, {
+  useMemo,
+  useCallback,
+  useRef,
+  useEffect,
+  useState,
+} from 'react'
+import { useHistory } from 'react-router-dom'
 
 import { useStore } from '../../contexts/Store'
-import NodeList from '../NodeList'
+import { useController } from '../../contexts/Controller'
+import usePaged from '../../hooks/usePaged'
+import { Node } from '../../utils/api/tracker'
+import useIsMounted from '../../hooks/useIsMounted'
 
+import NodeList from '../NodeList'
 import NodeStats from '../NodeStats'
 
 type Props = {
   id?: string
 }
 
-const TopologyList = ({ id }: Props) => {
-  const { nodes } = useStore()
-
-  const currentNode = useMemo(() => nodes.find(({ id: nodeId }) => nodeId === id), [nodes, id])
-  const { title } = (currentNode || {}).location || {}
-
-  return currentNode ? (
-    <NodeList>
-      <NodeList.Node
-        nodeId={currentNode.id}
-        title={currentNode.title}
-        address={currentNode.address}
-        placeName={title || ''}
-        isActive
-      >
-        <NodeStats key={currentNode.id} id={currentNode.address} />
-      </NodeList.Node>
+const TopologyList = ({ id: activeNodeId }: Props) => {
+  const { visibleNodes: nodes } = useStore()
+  const { loadNodeLocations } = useController()
+  const history = useHistory()
+  const listRef = useRef<HTMLDivElement>(null)
+  const scrollTimeout = useRef<number | undefined>(undefined)
+  const [pageChangedOnLoad, setPageChangedOnLoad] = useState(false)
+  const isMounted = useIsMounted()
+
+  const currentNode = useMemo(() => nodes.find(({ id: nodeId }) => (
+    nodeId === activeNodeId),
+  ), [nodes, activeNodeId])
+
+  const toggleNode = useCallback(
+    (nodeId) => {
+      history.replace(`/nodes/${encodeURIComponent(nodeId)}`)
+    },
+    [history],
+  )
+
+  // show all nodes in same location
+  const visibleNodes: Node[] = useMemo(() => ((currentNode && nodes) ? (
```

Comment:
True, it could pass the list of clicked nodes... It's a click handler so probably it won't work when visiting a node directly from url. Maybe it's possible but would require some more work during the loading phase 🤔